plugins/integrations/kubernetes-service/src/main/java/org/apache/cloudstack/api/command/user/kubernetes/version/ListKubernetesSupportedVersionsCmd.java

```diff
@@ -0,0 +1,113 @@
+        responseObject = KubernetesSupportedVersionResponse.class,
+        responseView = ResponseObject.ResponseView.Restricted,
+        authorized = {RoleType.Admin, RoleType.ResourceAdmin, RoleType.DomainAdmin, RoleType.User})
+public class ListKubernetesSupportedVersionsCmd extends BaseListCmd {
+    public static final Logger LOGGER = Logger.getLogger(ListKubernetesSupportedVersionsCmd.class.getName());
+    public static final String APINAME = "listKubernetesSupportedVersions";
+
+    @Inject
+    private KubernetesVersionService kubernetesVersionService;
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+    @Parameter(name = ApiConstants.ID, type = CommandType.UUID,
+            entityType = KubernetesSupportedVersionResponse.class,
+            description = "the ID of the Kubernetes supported version")
+    private Long id;
+
+    @Parameter(name = ApiConstants.ZONE_ID, type = CommandType.UUID,
+            entityType = ZoneResponse.class,
+            description = "the ID of the zone in which Kubernetes supported version will be available")
+    private Long zoneId;
+
+    @Parameter(name = ApiConstants.MIN_SEMANTIC_VERSION, type = CommandType.STRING,
+            description = "the minimum semantic version for the Kubernetes supported version to be listed")
+    private String minimumSemanticVersion;
+
+    @Parameter(name = ApiConstants.MIN_KUBERNETES_VERSION_ID, type = CommandType.UUID,
+            entityType = KubernetesSupportedVersionResponse.class,
+            description = "the ID of the minimum Kubernetes supported version")
+    private Long minimumKubernetesVersionId;
+
+    /////////////////////////////////////////////////////
+    /////////////////// Accessors ///////////////////////
+    /////////////////////////////////////////////////////
+    public Long getId() {
+        return id;
+    }
+
+    public Long getZoneId() {
+        return zoneId;
+    }
+
+    public String getMinimumSemanticVersion() {
+        if(!Strings.isNullOrEmpty(minimumSemanticVersion) &&
+                !minimumSemanticVersion.matches("[0-9]+(\\.[0-9]+)*")) {
+            throw new IllegalArgumentException("Invalid version format");
+        }
+        return minimumSemanticVersion;
+    }
+
+    public Long getMinimumKubernetesVersionId() {
+        return minimumKubernetesVersionId;
+    }
+
+    @Override
+    public String getCommandName() {
+        return APINAME.toLowerCase() + "response";
+    }
+
+    /////////////////////////////////////////////////////
+    /////////////// API Implementation///////////////////
+    /////////////////////////////////////////////////////
+    @Override
+    public void execute() throws ResourceUnavailableException, InsufficientCapacityException, ServerApiException, ConcurrentOperationException, ResourceAllocationException, NetworkRuleConflictException {
```

Comment:
these exceptions are unacceptable (intended pun miserably failed) we are listing clusters, no resource-, allocation-, or any change related exception should ever occur. at most a entity not found kind of thing.

Reply:
fixed